File: backend/app/core/upgrades/stat_modifiers/player_health_2.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class PlayerHealth2(UpgradeImplementation):
    id = 'player_health_2'
    name = 'Reinforced Core'
    description = '+30 max health'
    rarity = 'uncommon'
    type = 'stat_modifier'
    cost = 6

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        value = 30
        appliers.modifiers.add_multiplier('player', 'maxHealth', value)

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/stat_modifiers/player_health_2.ts
import { Upgrade, type UpgradeDef } from '../Upgrade'
import { UpgradeTargetID, UpgradeStatID, RarityID, UpgradeTypeID } from '../../data/ID'

export const PlayerHealth2Def: UpgradeDef = {
  id: "player_health_2",
  name: "Reinforced Core",
  description: "+30 max health",
  rarity: RarityID.Uncommon,
  upgradeType: UpgradeTypeID.StatModifier,
  cost: 6,
  targetClass: UpgradeTargetID.Player,
  fieldInTargetClass: UpgradeStatID.MaxHealth,
  value: 30,
  stackable: true,
  maxStacks: 99999999,
}

export class PlayerHealth2 extends Upgrade {}
